Amazon.Ion.ObjectMapper/IonSerializer.cs

```diff
@@ -172,87 +175,101 @@ public void Serialize<T>(IIonWriter writer, T item)
                 return;
             }
 
-            if (item is bool)
+            var type = item.GetType();
+
+            if (type.IsAssignableTo(typeof(bool)))
             {
-                new IonBooleanSerializer().Serialize(writer, Convert.ToBoolean(item));
+                var serializer = this.GetCustomPrimitiveSerializer<bool>() ?? new IonBooleanSerializer();
+                serializer.Serialize(writer, Convert.ToBoolean(item));
                 return;
             }
 
-            if (item is int)
+            if (type.IsAssignableTo(typeof(int)))

```

Comment:
Same goes for most of the other ones here, please check them. `is` is fine for almost all of these cases.

Reply:
Addressed.